presto-orc/src/main/java/com/facebook/presto/orc/OrcSelectiveRecordReader.java

```diff
@@ -428,6 +439,27 @@ private void initializePositions(int batchSize)
         }
     }
 
+    private int applyFilterFunctionsWithNoInputs(int[] positions, int positionCount)
+    {
+        initializeOutputPositions(positionCount);
+
+        for (FilterFunction function : filterFunctionsWithoutInputs) {
+            Page page = new Page(positionCount);
+            positionCount = function.filter(page, outputPositions, positionCount, errors);
+            if (positionCount == 0) {
+                break;
+            }
+        }
+
+        // at this point outputPositions are relative to page, e.g. they are indices into positions array
```

Comment:
Since no-input functions run first, this translation is not needed.

Reply:
That's right!